app/scripts/modules/core/pipeline/config/services/pipelineConfig.service.ts

```diff
@@ -135,6 +135,11 @@ export class PipelineConfigService {
     return uniq(upstreamStages);
   }
 
+  public startAdHocPipeline(body: any) {
```

Comment:
IMO not worth typing the body, since it is a JSON blob of various pipeline stages.